kafka-broker/src/main/scala/com/lightbend/lagom/internal/broker/kafka/Consumer.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.broker.kafka
+
+import java.util.concurrent.CompletionStage
+
+import scala.compat.java8.FutureConverters._
+import scala.util.{ Failure, Success }
+
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.slf4j.LoggerFactory
+
+import com.lightbend.lagom.internal.broker.kafka.serializer.KafkaDeserializer
+import com.lightbend.lagom.javadsl.api.Descriptor.TopicCall
+import com.lightbend.lagom.javadsl.api.ServiceInfo
+import com.lightbend.lagom.javadsl.api.broker.Subscriber
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.kafka.{ ConsumerMessage, ConsumerSettings, Subscriptions }
+import akka.kafka.scaladsl.{ Consumer => ReactiveConsumer }
+import akka.stream.{ FlowShape, Materializer }
+import akka.stream.javadsl.{ Flow => JFlow, Source => JSource }
+import akka.stream.scaladsl.{ Flow, GraphDSL, Sink, Unzip, Zip }
+
+/**
+ * A Consumer for consuming messages from Kafka using the akka-stream-kafka API. 
+ */
+class Consumer[Message] private (config: KafkaConfig, topicCall: TopicCall[Message], groupId: Subscriber.GroupId, info: ServiceInfo, system: ActorSystem)(implicit mat: Materializer) extends Subscriber[Message] {
+
+  private val log = LoggerFactory.getLogger(classOf[Consumer[_]])
+
+  @throws(classOf[Subscriber.IllegalGroupId])
+  override def withGroupId(groupId: String): Subscriber[Message] = {
+    val newGroupId = {
+      if (groupId == null) {
+        // An empty group id is not allowed by Kafka (see https://issues.apache.org/jira/browse/KAFKA-2648
+        // and https://github.com/akka/reactive-kafka/issues/155)
+        val defaultGroupId = Consumer.GroupId.default(info)
+        log.debug {
+          "Passed a null groupId, but Kafka requires clients to set one (see KAFKA-2648). " +
+            s"Defaulting $this consumer groupId to $defaultGroupId."
+        }
+        defaultGroupId
+      } else Consumer.GroupId(groupId)
+    }
+
+    if (newGroupId.groupId == groupId) this
+    else new Consumer(config, topicCall, newGroupId, info, system)
+  }
+
+  override def atMostOnceSource: JSource[Message, _] = {
```

Comment:
This is a good point, and I can't think of a use case where it could be useful. @jroper can you?